app/uk/gov/hmrc/cardpaymentfrontend/controllers/PaymentFailedController.scala

```diff
@@ -46,11 +46,11 @@ class PaymentFailedController @Inject() (
     ))
   }
 
-  val submit: Action[AnyContent] = Action { implicit request =>
+  val submit: Action[AnyContent] = actions.journeyAction { implicit journeyRequest: JourneyRequest[AnyContent] =>
     ChooseAPaymentMethodForm.form
       .bindFromRequest()
       .fold(
-        (formWithErrors: Form[ChooseAPaymentMethodForm]) => BadRequest(paymentFailedPage(taxType = "Self Assessment", true, formWithErrors)),
+        (formWithErrors: Form[ChooseAPaymentMethodForm]) => BadRequest(paymentFailedPage(taxType = journeyRequest.journey.taxType.toString, hasOpenBanking = true, form = formWithErrors)),
```

Comment:
the tax type is a string but used to compare with if it is "pngr". Don't use string, use a type such as taxType

Reply:
and update the twirl too